apps/admin/src/app/(api)/api/v1/anomaly/route.ts

```diff
@@ -0,0 +1,56 @@
+import { NextResponse, type NextRequest } from "next/server";
+import { isNoticeableEnergyConsumption } from "@/actions/energy-monitoring";
+import { env } from "@/env.mjs";
+import { getSensorsByUser } from "@/query/sensor";
+import { getAllUsers } from "@/query/user";
+
+import { getEnergyForSensorInRange } from "@energyleaf/db/query";
+import type { ConsumptionData } from "@energyleaf/lib";
+import { AggregationType } from "@energyleaf/lib";
+import { sendAnomalyEmail } from "@energyleaf/mail";
+
+export const POST = async (req: NextRequest) => {
+    const reportApiKey = env.REPORTS_API_KEY;
+    if (!req.headers.has("Authorization") || req.headers.get("Authorization") !== reportApiKey) {
+        return NextResponse.json({ status: 403, statusMessage: "Forbidden" });
+    }
+
+    const start = new Date(new Date().setHours(0, 0, 0, 0));
+    const end = new Date(new Date().setHours(23, 59, 59, 999));
+
+    try {
+        const users = await getAllUsers();
+        for (const user of users) {
+            const sensors = await getSensorsByUser(user.id);
+            let data: ConsumptionData[] = [];
+
+            for (const sensor of sensors) {
```

Comment:
Ich denke, wir sollten die ganze Annomalie-Erkennung so weit wie möglich auf der DB ausführen. Denn aktuell werden hier viele Daten in den Arbeitsspeicher geladen und dort schon intensive Operationen ausgeführt. Und es müssen außerdem eine Menge an Daten von der DB an unser Backend gesendet werden.
Bei der Berechnung verwendest du ja eh nur Standardabweichung bzw. Summe und Mittelwert. Diese Werte kann man ja aber auch direkt auf der DB ausführen. Theoretisch könnte man die DB-Abfrage dann so entwickeln, dass sie nur einen Boolean zurückgibt (ist signifikant oder nicht)